### PR TITLE
feat: assignment restriction

### LINKED ISSUE #1892: Assignment Restrictions
We have got quite a few complaints from professors that CircuitVerse allows cheating easily. For example users can easily copy/paste any public circuit. They can give collaborator access to their own circuit and copy paste etc. So it has been requested by professors to disable this feature. 

I think we need to optionally disable the following features
1. Combinational Analysis Tool
2. Verilog tools
3. Copy / Paste
4. Allow Collaborators to be added
5. Something else?

We need to find an elegant way to this from the assignment page. And also we need to be able to interface this nicely with the simulator. And also try and make it difficult for students to "hack" around this.  

**Are you working on this? (Yes/No)**
No

@Nitish145 @tachyons inputs here?

### PR DESCRIPTION
**Pull request by vedant-jain03**
Fixes #1892 

#### Describe the changes you have made in this PR -

- added new column `feature_restrictions` in `assignment` table of type `jsonb` with default `{}`
- added `restrictedFeature.js` will be used to display the checkbox in the assignment form.
- added a hidden field that appends the selected elements into form submission.
- added feature_restrictions in `params.permit`
- if the project has a restriction of 'add collaborators' or 'copy/paste' the link will have `display:none`
- if the project has a restriction of 'Verilog' or 'Combinational analysis tool' the link will have `display:none` in the simulator navbar. 

## Screenshot

1. Enable-Feature Restriction
Screenshot from 2022-07-19 19-03-33

2. Feature-Restrictions Section in Assignment
Screenshot from 2022-07-19 19-04-58

> Copy / Paste,Allow Collaborators is restricted hence the buttons are not being rendered in the Page.

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

**Comment by vedant-jain03** [comment]
@tachyons @satu0king Kindly Review this PR.
If there is anything I need to add or modify.

File: app/javascript/controllers/assignment_controller.js
Comment (vedant-jain03) [reply app/javascript/controllers/assignment_controller.js:49]: expected

File: app/javascript/controllers/assignment_controller.js
Comment (vedant-jain03) [reply app/javascript/controllers/assignment_controller.js:39]: expected

File: app/javascript/controllers/assignment_controller.js
Comment (vedant-jain03) [reply app/javascript/controllers/assignment_controller.js:15]: expected

**Comment by vedant-jain03** [comment]
@tachyons

File: app/decorators/assignment_decorator.rb
Comment (tachyons) [inline app/decorators/assignment_decorator.rb:37]:
```suggestion
     assignment.feature_restrictions.presence || "None"
```

**Review by tachyons — APPROVED** [review]
LGTM, can be merged once the UX is approved by @nitin10s

**Review by nitin10s — APPROVED** [review]
LGTM (Remove the screenshot as well)

**Comment by vedant-jain03** [comment]
cc @tachyons

**Pull request merged**
